TM1py/Services/CellService.py

```diff
@@ -368,6 +370,57 @@ def write_dataframe(self, cube_name: str, data: 'pd.DataFrame', dimensions: Iter
                           use_changeset=use_changeset,
                           **kwargs)
 
+    @require_pandas
+    @manage_transaction_log
+    def write_dataframe_async(self, cube_name: str, data: 'pd.DataFrame',
+                              slice_size_of_dataframe: int, max_workers: int,
```

Comment:
`slice_size_of_dataframe` could also be converted into a default argument and if nothing is passed, the method should intelligently split the DataFrame into multiple chunks especially when the number of records are less than 16380.

Reply:
I agree it would be nice to have default values, or `None` as default values and intelligence inside the function to determine sensible values considering the size of the passed data frame.

The value for `slice_size_of_dataframe` should be 32k or a multiple of 32k IMO. Since we squeeze 16k in the prolog and 16k in the epilog in the `write_through_unbound_process` function. So effectively TM1 consumes chunks of 32k.